.github/workflows/lint.yml

```diff
@@ -0,0 +1,24 @@
+name: lint
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  lint:
+    timeout-minutes: 10
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.9"
+
+      - name: Install Python Dependecies
+        working-directory: backend-py
+        run: pip install "pre-commit==2.20.0"
+
+      - name: Run Linting
+        run: pre-commit run --all-files
```

Comment:
We could further optimize this so we don't run against the whole repo but this is a good start.